libraries/dotnet/EnergyOrigin.Setup/EnergyOrigin.Setup/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using System.Text.Json.Serialization;
+using Asp.Versioning;
+using Asp.Versioning.ApiExplorer;
+using EnergyOrigin.Setup.Swagger;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace EnergyOrigin.Setup;
+
+public static class IServiceCollectionExtensions
+{
+    public static void AddSwagger(this IServiceCollection services, string subsystemName)
+    {
+        services.AddTransient<IConfigureOptions<SwaggerGenOptions>>(s =>
+            new ConfigureSwaggerOptions(s.GetRequiredService<IApiVersionDescriptionProvider>(), subsystemName));
+    }
+
+    public static void AddApiVersioning(this IServiceCollection services)
+    {
+        services.AddApiVersioning(options =>
+            {
+                options.AssumeDefaultVersionWhenUnspecified = false;
+                options.ReportApiVersions = true;
+                options.ApiVersionReader = new HeaderApiVersionReader("EO_API_VERSION");
+            })
+            .AddMvc()
+            .AddApiExplorer();
+    }
+
+    public static void AddControllersWithEnumsAsStrings(this IServiceCollection services)
+    {
+        services.AddControllers()
+            .AddJsonOptions(o =>
+                o.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter()));
+    }
+}
```

Comment:
If this is added to the NuGet, are we then forcing everyone who imports it, to use versioning in our api?

I suggest having a boolean as a featureflag, that makes one able toggle on/off, each feature that is included in the NuGet. Be it MassTransit, Swagger, JsonEnumStringConverter, etc.

Reply:
No only if you use the specified function for versioning. I would rather make a new function for without versioning, if that is needed, than making a boolean. But do we have apis that does not need versioning?